presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -482,7 +505,8 @@ private static Optional<Subfield> toSubfield(
                 RowExpression expression,
                 StandardFunctionResolution functionResolution,
                 ExpressionOptimizer expressionOptimizer,
-                ConnectorSession connectorSession)
+                ConnectorSession connectorSession,
+                Metadata metadata)
```

Comment:
can we pass functionAndTypeManager or functionAndTypeResolver instead of metadata?